src/shvec.c

```diff
@@ -36,18 +38,38 @@ int shvec_create(){
 }
 
 /*
-* Appends a value to the shvec, reallocates the array if necessary
+ * Expand memory allocation of shvec's data.
+ */
+int shvec_expand(int id) {
+    // Catch int overflow before changing max_size
+    while (shvec_array[id].size < shvec_array[id].max_size) {
```

Comment:
either move line 52 into the while or get rid of the while.

This way, the while is redundant AND can have the program get stuck in an infinite loop